homeassistant/components/zwave_js/addon.py

```diff
@@ -0,0 +1,256 @@
+"""Provide add-on management."""
+from __future__ import annotations
+
+import asyncio
+from functools import partial
+from typing import Callable, Optional, cast
+
+from awesomeversion import AwesomeVersion
+
+from homeassistant.components.hassio import (
+    async_get_addon_discovery_info,
+    async_get_addon_info,
+    async_install_addon,
+    async_set_addon_options,
+    async_start_addon,
+    async_stop_addon,
+    async_uninstall_addon,
+    async_update_addon,
+)
+from homeassistant.components.hassio.handler import HassioAPIError
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryNotReady, HomeAssistantError
+from homeassistant.helpers.singleton import singleton
+
+from .const import (
+    ADDON_SLUG,
+    CONF_ADDON_DEVICE,
+    CONF_ADDON_NETWORK_KEY,
+    CONF_NETWORK_KEY,
+    CONF_USB_PATH,
+    DOMAIN,
+    LOGGER,
+)
+
+DATA_ADDON_MANAGER = f"{DOMAIN}_addon_manager"
+
+
+@singleton(DATA_ADDON_MANAGER)
+@callback
+def get_addon_manager(hass: HomeAssistant) -> AddonManager:
+    """Get the add-on manager."""
+    return AddonManager(hass)
+
+
+async def async_ensure_addon_running(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Ensure that Z-Wave JS add-on is installed and running."""
+    addon_manager: AddonManager = get_addon_manager(hass)
+    try:
+        addon_is_installed = await addon_manager.async_is_addon_installed()
+        addon_is_running = await addon_manager.async_is_addon_running()
+    except AddonError as err:
+        LOGGER.error("Failed to get the Z-Wave JS add-on info")
+        raise ConfigEntryNotReady from err
```

Comment:
It's weird that this would raise a `ConfigEntryNotReady`. Then such functions should live in `__init__.py`.